src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -618,9 +621,13 @@ private void ExecuteClusterTransactionOneByOne(List<ClusterTransactionCommand.Si
                 var mergedCommand = new BatchHandler.ClusterTransactionMergedCommand(this, singleCommand);
                 try
                 {
+                    //The batch list only includes command with database commands.
+                    //We want to update also indexes for cluster trx commands without documents commands (has only cmpXchg commands)
+                    //So we update here until the index before the current command (and before executing it)  
+                    ClusterTransactionIndexWaiter.SetAndNotifyListenersIfHigher(command.Index - 1);
```

Comment:
I would prefer a simpler code and single code path.. it there any reason beyond "optimization" for this not be part of `OnClusterTransactionCompletion`?

Reply:
I don't want to send empty commands to the transaction merger.
It is not optimization but a clean way to code that.
Cluster transaction command with only compare exchange modification is already completed in the cluster state machine but we want to update the index according that only after all the `ClusterTransactionCommand`s with lower index are completed.
Here we know all `ClusterTransactionCommand`s equal to "command.Index - 1" and lower were completed